lib/precompiler.js

```diff
@@ -244,12 +258,19 @@ module.exports.cli = function(opts) {
   output.map = output.map + '';
 
   if (opts.min) {
-    output = uglify.minify(output.code, {
-      fromString: true,
+    if (uglifyExists()) {
+      // We are using `require` instead of `import` here, because es6-modules do not allow
+      // dynamic imports and uglify-js is an optional dependency.
+      let uglify = require('uglify-js');
+      output = uglify.minify(output.code, {
+        fromString: true,
+        outSourceMap: opts.map,
+        inSourceMap: JSON.parse(output.map)
+      });
+    } else {
+      console.error('Code minimization is disabled due to missing uglify-js dependency');
+    }
```

Comment:
why not just use try/catch here directly instead of if/else and the function above?

Reply:
I want to use require.resolve to detect if uglify exists, beause it wont be irritated by other missing-module-errors (caused by dependencies of uglify).

I could put the `.minify`-call into the try-catch-block, but that would be something like 

A function seems to be the best way to create a try-catch construct 

```js
let exists;
try {
  require.resolve('uglify-js');
  exists = true
} catch (e) {
  if (e.code !== 'MODULE_NOT_FOUND') {
    throw e;
  }
  exists = false
}
if (exists) {
  require('uglify-js').minify(...)
}
```

This seems to be more complicated and I don't particular like the use of the `exists` variable in this code. Creating a function seems to be more readable.
But I think I will still make a change.